dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -678,6 +679,14 @@ public void updateTrackedEntityInstancesSyncTimestamp( List<String> trackedEntit
         query.executeUpdate();
     }
 
+    @Override
+    public List<TrackedEntityInstance> getTrackedEntityInstancesByUid( List<String> uids, User user )

```

Comment:
Are we sure that we page the number of TEIs somehow? There is a max query size limit in PostgreSQL and it might explode here.

Reply:
This method is currently used by the AbstractRelationshipService, when it is preparing the cache before performing create and/or update. When the top-level method is called, it is partitioning the number of relationships to add/update each round, limiting this query to retrieve maximum 2x the partition size (one for each side), and that only happens if every single relationship only has tei on each side of the relationship, and each tei is unique :) I think it is fine for now?